### PR TITLE
add mse loss

### PR DESCRIPTION
**Pull request by MARD1NO**
增加 MSE Loss

**主要是调用已有的 squared_difference算子，结合reduce_mean / sum 操作**

有个问题是，reduce_mean后，**还是一个Tensor**（我需要用index，即x[0]获取其中的值）， 而**Numpy类似返回一个标量**，相关处理可以在我的单测脚本看到

另外有个单测问题是**如果是全随机数**，有时候oneflow 与 Numpy 计算**有些许误差，非常非常小**，测试到一次误差为 0.00000003。
如果想要以随机数进行测试，那么后续脚本得换成 np.allclose

现在是**输入数据写死的情况**，脚本采用 np.array_equal

TODO： 
- [x] 公式完善
- [x] 参数位置重新排一下

求导如下：
图片

### CODE REVIEW

File: oneflow/python/ops/nn_ops.py [hunk 1]
```diff
@@ -2646,3 +2646,114 @@ def leaky_relu_Job(x: tp.Numpy.Placeholder((5, ),)
...
+
+    Example 1: 
+
+    .. code-block:: python 
+
+        @flow.global_function()
+        def mseloss_job(input: tp.Numpy.Placeholder(shape=(3, 3)), 
+                        target: tp.Numpy.Placeholder(shape=(3, 3)))->tp.Numpy: 
+            # It's equal to `Torch.nn.MSELoss(reduction="mean")
+            out = flow.nn.MSELoss(input, target, reduction="mean")
```
Comment (doombeaker) [inline oneflow/python/ops/nn_ops.py:2699]: 注释忘记去掉了？下同

File: oneflow/python/test/ops/test_mse_loss.py [hunk 2]
```diff
@@ -0,0 +1,163 @@
+"""
```
Comment (doombeaker) [inline oneflow/python/test/ops/test_mse_loss.py:1]: 文件名用 `test_mseloss.py` 可能更符合测试目录中的习惯

File: oneflow/python/test/ops/test_mse_loss.py [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+def _compare_mseloss_with_np(
+    input_shape, target_shape, device_type, machine_ids, device_counts
+):
+    input = np.random.random(size=input_shape).astype(np.float32)
+    target = np.random.random(size=target_shape).astype(np.float32)
+
+    assert device_type in ["cpu", "gpu"]
+
+    flow.clear_default_session()
+    flow.env.init()
```
Comment (doombeaker) [inline oneflow/python/test/ops/test_mse_loss.py:34]: 这个 env.init 可以试着去掉，应该是没有影响的

File: oneflow/python/test/ops/test_mse_loss.py [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+        np_mse_sum = np.sum(np_mse)
+
+        return {
+            "np_mse_loss": np_mse,
+            "np_mse_loss_mean": np_mse_mean,
+            "np_mse_loss_sum": np_mse_sum,
+        }
+
+    def np_mseloss_grad(np_input, np_target):
+        elem_cnt = np_input.size
```
Comment (doombeaker) [inline oneflow/python/test/ops/test_mse_loss.py:54]: 这个 `elem_cnt` 只被一次用到的话，其实可以去掉，直接用在下面代码的 ` (-2 * (np_target - np_input)) / np_input.size` 中

File: oneflow/python/test/ops/test_mse_loss.py [hunk 5]
```diff
@@ -0,0 +1,163 @@
...
+                shape=input.shape,
+                dtype=flow.float32,
+                initializer=flow.ones_initializer(),
+                name="x_var",
+            )
+            x_var = of_input + v
+
+        flow.watch_diff(x_var, assert_prediction_grad)
+
+        mseloss = flow.nn.MSELoss(x_var, of_target, reduction="none", name="of_mseloss")
```
Comment (doombeaker) [inline oneflow/python/test/ops/test_mse_loss.py:91]: 为什么这里的reduction模式是"none"，然后梯度比较是和模拟的"mean"梯度比较？

File: oneflow/python/test/ops/test_mse_loss.py [hunk 6]
```diff
@@ -0,0 +1,163 @@
...
+
+
+@flow.unittest.skip_unless_1n1d()
+class Testl1loss1n1d(flow.unittest.TestCase):
+    def test_mseloss_cpu(test_case):
+        arg_dict = _gen_arg_dict(
+            shape=(3, 16), device_type="cpu", machine_ids="0:0", device_counts=1
+        )
+
+        for arg in GenArgList(arg_dict):
```
Comment (doombeaker) [inline oneflow/python/test/ops/test_mse_loss.py:141]: 只有一个样本，不需要for了。直接 `_compare_mseloss_with_np(list(arg_dict.values())` 或者 `_compare_mseloss_with_np(list(GenArgList(arg_dict)[0])`

想保留现在循环形式，方便以后添加测试sample也可以，自己权衡下吧

File: oneflow/python/test/ops/test_mse_loss.py (hunk 4 above)
Comment (MARD1NO) [reply oneflow/python/test/ops/test_mse_loss.py:54]: 这个我觉得单独起一个变量，代码会比较容易读

File: oneflow/python/test/ops/test_mse_loss.py (hunk 6 above)
Comment (MARD1NO) [reply oneflow/python/test/ops/test_mse_loss.py:141]: 我倾向于保留，方便后续添加测试

**Review by doombeaker — APPROVED** [review]
